src/app/admin/products/[productId]/edit/page.tsx

```diff
@@ -0,0 +1,779 @@
+"use client";
+import { useParams } from "next/navigation";
+import { doc, getDoc } from "firebase/firestore";
+import { db } from "../../../../../utils/firebase";
+import { useEffect, useState, useActionState } from "react";
+import { Button } from "../../../../components/Button";
+import {
+  allCategories,
+  allTags,
+  allAvailabilityStatus,
+  allReturnPolicies,
+  Product,
+} from "../../../../../types/types";
+import { editProduct } from "../../../../actions/admin/products";
+import Form from "next/form";
+import { useForm } from "react-hook-form";
+
+const initialState: EditProductFormState = {
+  success: false,
+  inputs: {},
+  errors: {},
+};
+
+export interface EditProductFormState {
+  success: boolean;
+  message?: string;
+  inputs?: Partial<Product>;
+  errors?: {
+    [K in keyof Product]?: string[];
+  };
+  data?: any;
+}
+
+export default function EditProduct() {
+  const {
+    register,
+    formState: { errors },
+  } = useForm({ mode: "all" });
+  const [state, formAction, isPending] = useActionState<
+    EditProductFormState,
+    FormData
+  >(editProduct, initialState);
+
+  const params = useParams<{ productId: string }>();
+  const [product, setProduct] = useState<any>();
+
+  useEffect(() => {
+    async function fetchData() {
+      const docRef = doc(db, "products", params.productId);
+      const docSnap = await getDoc(docRef);
+      if (docSnap.exists()) {
+        setProduct(docSnap.data());
+      } else {
+        console.log("No such document!");
```

Comment:
Display an error message on the screen stating that the product is not found and include a link to return to the previous page.